network/src/peer_registry.rs

```diff
@@ -81,11 +80,16 @@ impl PeerRegistry {
         if self.peers.contains_key(&session_id) {
             return Err(PeerError::SessionExists(session_id).into());
         }
+        let peer_id = extract_peer_id(&remote_addr).expect("opened session should have peer id");
         if self.get_key_by_peer_id(&peer_id).is_some() {
             return Err(PeerError::PeerIdExists(peer_id).into());
         }
 
-        let is_whitelist = self.whitelist_peers.contains(&peer_id);
+        let is_whitelist = self.whitelist_peers.iter().any(|whitelist_peer| {
```

Comment:
See the previous comment.

Reply:
I think we should also check whether an address has a peer_id in this function and reject all address which hasn't have a peer_id. Thus, we can keep the behavior consistent.